<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SetParameter</class>
 <widget class="QWidget" name="SetParameter">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>672</width>
    <height>485</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="2,3">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4" stretch="1,1">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string/>
         </property>
         <layout class="QGridLayout" name="gridLayout_14">
          <item row="0" column="0">
           <widget class="QComboBox" name="comboBox_baude">
            <property name="minimumSize">
             <size>
              <width>80</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="pushButton_set">
            <property name="text">
             <string>设置CAN波特率</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QComboBox" name="comboBox_model">
            <property name="minimumSize">
             <size>
              <width>80</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>正常模式</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>自测模式</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="pushButton_model">
            <property name="text">
             <string>设置模式</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QPushButton" name="pushButton_reset">
            <property name="text">
             <string>复位CAN</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QPushButton" name="pushButton_reset_error">
            <property name="text">
             <string>错误复位</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0" colspan="2">
           <widget class="QPushButton" name="pushButton_read_can">
            <property name="text">
             <string>读取CAN参数</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string/>
         </property>
         <layout class="QGridLayout" name="gridLayout_15">
          <item row="0" column="0">
           <widget class="QLabel" name="label_39">
            <property name="maximumSize">
             <size>
              <width>45</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>波特率:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="comboBox_baude_par">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="currentIndex">
             <number>-1</number>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="label_38">
            <property name="text">
             <string>停止位:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QComboBox" name="comboBox_stopbits_par">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>1</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1.5</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_43">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>数据位:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="comboBox_databits_par">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>0</height>
             </size>
            </property>
            <property name="currentIndex">
             <number>3</number>
            </property>
            <item>
             <property name="text">
              <string>5</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>6</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>7</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>8</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLabel" name="label_41">
            <property name="maximumSize">
             <size>
              <width>45</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>流控制:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QComboBox" name="comboBox_flowControl_par">
            <property name="minimumSize">
             <size>
              <width>80</width>
              <height>0</height>
             </size>
            </property>
            <property name="currentIndex">
             <number>1</number>
            </property>
            <item>
             <property name="text">
              <string>硬件流控制</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>无流控制</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_42">
            <property name="text">
             <string>模式:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="comboBox_model_par">
            <property name="minimumSize">
             <size>
              <width>60</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>RS232</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>RS485</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>RS422</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QLabel" name="label_40">
            <property name="maximumSize">
             <size>
              <width>45</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>校验位:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QComboBox" name="comboBox_parity_par">
            <property name="minimumSize">
             <size>
              <width>80</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>无校验</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>奇校验</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>偶校验</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>强制为0</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>强制为1</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QPushButton" name="pushButton_com_par">
            <property name="text">
             <string>设置串口参数</string>
            </property>
           </widget>
          </item>
          <item row="3" column="2" colspan="2">
           <widget class="QPushButton" name="pushButton_com_read_par">
            <property name="text">
             <string>串口读取MCU参数</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string/>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <widget class="QLabel" name="label_21">
            <property name="maximumSize">
             <size>
              <width>105</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>转换模式:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="comboBox_ts_model">
            <item>
             <property name="text">
              <string>透明转换</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>透明带标识转换</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>格式转换</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>MODBUS转换</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_20">
            <property name="maximumSize">
             <size>
              <width>160</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>转换方向:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="comboBox_ts_direct">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>双向转换</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>仅串口转到CAN</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>仅CAN转到串口</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_23">
            <property name="maximumSize">
             <size>
              <width>105</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>串行帧间隔(ms):</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="spinBox_frame_times">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>250</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_25">
            <property name="maximumSize">
             <size>
              <width>160</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>CAN帧转发间隔(ms):</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSpinBox" name="spinBox_ts_times">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>250</number>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_22">
            <property name="maximumSize">
             <size>
              <width>105</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>CAN帧ID转换设置:</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QComboBox" name="comboBox_ts_id">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>不允许</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>允许</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_19">
            <property name="maximumSize">
             <size>
              <width>105</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>CAN帧信息转换设置:</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QComboBox" name="comboBox_ts_msg">
            <property name="minimumSize">
             <size>
              <width>70</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>不允许</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>允许</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_18">
            <property name="maximumSize">
             <size>
              <width>160</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>标识符在串行帧中的起始偏移:</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1">
           <widget class="QComboBox" name="comboBox_frame_offset">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>0</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>1</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>3</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>4</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>5</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>6</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>7</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="7" column="0">
           <widget class="QLabel" name="label_24">
            <property name="maximumSize">
             <size>
              <width>160</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>标识符在串行帧中的长度:</string>
            </property>
           </widget>
          </item>
          <item row="7" column="1">
           <widget class="QComboBox" name="comboBox_frame_len">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>1</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>3</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>4</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="8" column="1">
           <widget class="QPushButton" name="pushButton_ts">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>设置转换参数</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string/>
         </property>
         <layout class="QGridLayout" name="gridLayout_10">
          <item row="0" column="0">
           <widget class="QLabel" name="label_15">
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>CAN通信波特率:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="comboBox_baude_can">
            <property name="minimumSize">
             <size>
              <width>70</width>
              <height>0</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_16">
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>CAN帧ID:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="lineEdit_can_id">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>00 00 00 01</string>
            </property>
            <property name="placeholderText">
             <string>00 00 00 01</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_17">
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>帧格式:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="comboBox_frame_format">
            <property name="minimumSize">
             <size>
              <width>70</width>
              <height>0</height>
             </size>
            </property>
            <item>
             <property name="text">
              <string>标准帧</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>扩展帧</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QPushButton" name="pushButton_set_can">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>设置CAN参数</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
